src/Forms/SearchableDropdownTrait.php

```diff
@@ -0,0 +1,594 @@
+     */
+    public function setSearchContext(?SearchContext $searchContext): static
+    {
+        $this->searchContext = $searchContext;
+        $this->setUseSearchContext(true);
+        return $this;
+    }
+
+    /**
+     * Get whether to use a dynamic placeholder if a normal placeholder is not set
+     */
+    public function getUseDynamicPlaceholder(): bool
+    {
+        return $this->useDynamicPlaceholder;
+    }
+
+    /**
+     * Set whether to use a dynamic placeholder if a normal placeholder is not set
+     */
+    public function setUseDynamicPlaceholder(bool $useDynamicPlaceholder): static
+    {
+        $this->useDynamicPlaceholder = $useDynamicPlaceholder;
+        return $this;
+    }
+
+    /**
+     * Get whether to use a search context instead searching on labelField
+     */
+    public function getUseSearchContext(): bool
+    {
+        return $this->useSearchContext;
+    }
+
+    /**
+     * Set whether to use a search context instead searching on labelField
+     */
+    public function setUseSearchContext(bool $useSearchContext): static
+    {
+        $this->useSearchContext = $useSearchContext;
+        return $this;
+    }
+
+    /**
+     * Get whether the field allows searching by typing characters into field
+     */
+    public function getIsSearchable(): bool
+    {
+        return $this->isSearchable;
+    }
+
+    /**
+     * Set whether the field allows searching by typing characters into field
+     */
+    public function setIsSearchable(bool $isSearchable): static
+    {
+        $this->isSearchable = $isSearchable;
+        return $this;
+    }
+
+    /**
+     * This returns an array rather than a DataList purely to retain compatibility with ancestor getSource()
+     */
+    public function getSource(): array
+    {
+        return $this->getListMap($this->sourceList);
+    }
+
+    /*
+     * @param mixed $source
+     */
+    public function setSource($source): static
+    {
+        if (!is_a($source, DataList::class)) {
+            throw new InvalidArgumentException('Source must be a DataList');
+        }
+        // Setting to $this->sourceList instead of $this->source because SelectField.source
+        // docblock type is array|ArrayAccess i.e. does not allow DataList
+        $this->sourceList = $source;
+        return $this;
+    }
+
+    /**
+     * Get the field to use for the label of the option
+     *
+     * The default value of 'Title' will map to DataObject::getTitle() if a Title DB field does not exist
+     */
+    public function getLabelField(): string
+    {
+        return $this->labelField;
+    }
+
+    /**
+     * Set the field to use for the label of the option
+     */
+    public function setLabelField(string $labelField): static
+    {
+        $this->labelField = $labelField;
+        return $this;
+    }
+
+    public function getAttributes(): array
+    {
+        $name = $this->getName();
+        if ($this->getIsMultiple() && strpos($name, '[') === false) {
```

Comment:
Why would the name already have square brackets? That seems like it would be someone doing something wrong and possibly shouldn't be catered for, unless I'm missing something?

Reply:
Just did a quick debug, when rendering a SearchMultiDropdownField on a Page edit form for a many_many relation on a Page, the return value from $this->getName() didn't have square brackets

Reply:
Right, so if it doesn't have square brackets we don't need the `strpos` check - just always add them on.

Reply:
Done